opm/autodiff/BlackoilWellModel.hpp

```diff
@@ -153,6 +153,16 @@ namespace Opm {
             const SimulatorReport& lastReport() const;
 
 
+            void addWellContributions(Mat& mat)
+            {
+                for (int w = 0; w < numWells(); ++w) {
+                    auto* well = dynamic_cast<StandardWell<TypeTag>*>(well_container_[w].get());
```

Comment:
This line is a hack, and must be changed (I know it comes from my own commit...), re-adding addWellContributions() as an interface method would work.

The situation is also somewhat confusing with respect to what matrix is used where, and it would be good for testing to still have the possibility to also use the "full" matrix for the linear operator. I suggest adding another bool parameter controlling that.